src/render/src/macros/structure.rs

```diff
@@ -18,8 +18,20 @@
 macro_rules! gfx_impl_struct {
     ($runtime_format:ty : $compile_format:path = $root:ident {
         $( $field:ident: $ty:ty = $name:expr, )*
+    }) => (gfx_impl_struct_meta! {
+        impl_struct_meta $runtime_format : $compile_format = $root {
+            $( $field : $ty = $name, )*
+        }
+    })
+}
+
+#[macro_export]
+macro_rules! gfx_impl_struct_meta {
```

Comment:
I could probably have two patterns inside `gfx_impl_struct`, one with `$(#[$attr:meta])* impl_struct_meta` and one without, but then I believe I'd have to duplicate the whole body, which is worse to me than having an extra name.